aptos-move/move-examples/dao/nft_dao/sources/nft_dao.move

```diff
@@ -0,0 +1,956 @@
+
+        // assert the proposal voting ended
+        let proposals = borrow_global_mut<Proposals>(nft_dao);
+        let proposal = table::borrow_mut(&mut proposals.proposals, proposal_id);
+        let now = timestamp::now_seconds();
+        assert!(now >= proposal.start_time_sec + dao.voting_duration, error::invalid_argument(EPROPOSAL_ENDED));
+
+        // assert the proposal is unresolved yet
+        assert!(proposal.resolution == PROPOSAL_PENDING, error::invalid_argument(EPROPOSAL_RESOLVED));
+
+        let proposal_stat = &mut borrow_global_mut<ProposalVotingStatistics>(nft_dao).proposals;
+        let voting_stat = table::borrow_mut(proposal_stat, proposal_id);
+        proposal.final_yes_votes = voting_stat.total_yes;
+        proposal.final_no_votes = voting_stat.total_no;
+        // validate resolve threshold and result
+        let voted = voting_stat.total_no + voting_stat.total_yes;
+        if (voted < dao.resolve_threshold) {
+            // not sufficient token voted
+            proposal.resolution = PROPOSAL_RESOLVED_NOT_PASSED;
+        } else if(voting_stat.total_yes > voting_stat.total_no) {
+            let function_name = proposal.function_name;
+            if (function_name == string::utf8(b"transfer_fund")) {
+                let res_signer =  create_signer_with_capability(&dao.dao_signer_capability);
+                let dst_addr = property_map::read_address(&proposal.function_args, &string::utf8(b"dst"));
+                let amount = property_map::read_u64(&proposal.function_args, &string::utf8(b"amount"));
+                transfer_fund<CoinType>(&res_signer, dst_addr, amount);
+            } else {
+               assert!(function_name == string::utf8(b"no_op"), error::invalid_argument(ENOT_SUPPROTED_FUNCTION));
+            };
+            proposal.resolution = PROPOSAL_RESOLVED_PASSED;
+        } else {
+            proposal.resolution = PROPOSAL_RESOLVED_NOT_PASSED;
+        };
+
+        nft_dao_events::emit_resolve_event(
+            proposal_id,
+            nft_dao,
+            proposal.resolution,
+        )
+    }
+
+    /// Offer DAO to an address
+    public entry fun offer_dao(admin: &signer, dao: address, new_admin: address) acquires DAO, OwnedDAO, DAOOffer {
+        let admin_addr = signer::address_of(admin);
+        let dao_config = borrow_global<DAO>(dao);
+        assert!(admin_addr == dao_config.admin, error::permission_denied(EINVALID_ADMIN_ACCOUNT));
+        let owned_dao = &borrow_global<OwnedDAO>(admin_addr).daos;
+        let (found, _) = vector::index_of(owned_dao, &dao);
+        assert!(found, error::not_found(ENOT_IN_ADMIN_OWNED_DAO_ADDRESSES));
+
+        if(!exists<DAOOffer>(admin_addr)){
+            let tb = table::new();
+            let dst = vector::empty();
+            vector::push_back(&mut dst, dao);
+            table::add(&mut tb, new_admin, dst);
+            move_to(admin, DAOOffer {
+                offers: tb,
+            });
+        } else {
```

Comment:
This looks very complex. If we store the pending transfer in the DAO account, this can be simplified a lot. No table or vector needed.

Reply:
good idea. will revise